### DISCUSSION TITLE
Model Database Integrity Constraint  prevents Model deletion but not its media

### DISCUSSION THREAD

**Discussion by asolopovas** [spatie/laravel-medialibrary#2661] [Q&A]
Hi, 

I have a `Product` model which has media attached to it which also has the database constraint that prevents its deletion if this product is used in the order. If I try to remove the product that is used in the order it throws an exception and remains intact but the media gets deleted anyway for some reason.

laravel-medialibrary version: 9.8.3

**Comment by freekmurze** [comment]
Please submit a test for this for us to fix.

**Reply by asolopovas** [reply]
If I change     `protected bool $deletePreservingMedia = false;` to `true` in InterctsWithMedia class it works without problems, but with default option set to false it deletes the media, is there a way I can change this default without actually changing the value inside the package?

**Comment by erikn69** [comment]
Are you using transactions? 
because on exception you must do a rollback and everything are restored

**Reply by asolopovas** [reply]
I am not using transactions, I am using lighthouse-php (graphql server) whenever I trigger delete mutation it prevents the deletion of model, but deletes the media of that model from database as well as storage folder
